src/java.desktop/share/classes/javax/swing/plaf/basic/BasicPopupMenuUI.java

```diff
@@ -1226,6 +1226,7 @@ public void stateChanged(ChangeEvent ev) {
                 // menu hidden -- return focus to where it had been before
                 // and uninstall menu keybindings
                    removeItems();
+                   menuInputMap = null;
```

Comment:
Just curious, will the uninstall(); at the start of the method will clear this flag as well?

Reply:
We do call ninstall() at the beginning of the method but it does not resolve the issue.

Reply:
I meant this code at the start of the method:
`            if (!(UIManager.getLookAndFeel() instanceof BasicLookAndFeel)) {
                uninstall();
                return;
            }`
If "uninstall" will not solve the problem and the code added in this fix will not be executed, then we will get the same leak?

Reply:
The code block here with uninstall is called if currently installed LAF is not derived from BasicLAF in which case we will not have this problem in the first place - the only place where we are setting menuInputMap variable is in this method down the line so it never gets initialized. And no, calling uninstall() unconditionally will not solve the problem because uninstall() does not change this value either.

Reply:
I guess that the uninstall() above is called when the L&F was basic(since BasicPopupMenuUI is used) and then changed for some reason, so when the method in the BasicPopupMenuUI.java will be called we will find that the current L&F is different and call "uninstall()", so this is my question: will the leak will exist in that case? I mean the code changed by the current fix will not be executed since we will exit after "uninstall()".